File: bookmark-manager-backend/Models/Bookmark.cs
using System.Text.Json.Serialization;

namespace BookmarkManager.Models
{
    public class Bookmark
    {
        [JsonPropertyName("id")]
        public int Id { get; set; }
        
        [JsonPropertyName("link")]
        public string Url { get; set; } = string.Empty;
        
        [JsonPropertyName("title")]
        public string? Title { get; set; } = string.Empty;
        
        [JsonPropertyName("created_at")]
        public DateTime CreateDate { get; set; }
        
        [JsonPropertyName("read_date")]
        public DateTime? ReadDate { get; set; }
    }
}

File: bookmark-manager-frontend/src/models/Bookmark.ts
export interface Bookmark {
  id: number;
  url?: string;
  link?: string;  // Backend uses 'link' instead of 'url'
  title: string | null;
  created_at?: string;  // Backend uses 'created_at' for the creation date
  read_date: string | null;
}